### PR TITLE
Fix code smells

### PR DESCRIPTION
**Pull request by zyoneal**

### CODE REVIEW

File: api/src/main/java/com/eleks/academy/pharmagator/controllers/MedicineController.java [hunk 1]
```diff
@@ -49,7 +48,7 @@ public MedicineDto create(@Valid @RequestBody MedicineDto medicineDto) {
     }
 
     @DeleteMapping("/{id:[\\d]+}")
-    public ResponseEntity<?> delete(@PathVariable Long id) {
+    public ResponseEntity<Object> delete(@PathVariable Long id) {
```
Comment (Oleksandr-Eleks) [inline api/src/main/java/com/eleks/academy/pharmagator/controllers/MedicineController.java:51]: I would suggest to use here MedicineDto

File: api/src/main/java/com/eleks/academy/pharmagator/controllers/PharmacyController.java [hunk 2]
```diff
@@ -49,7 +48,7 @@ public PharmacyDto create(@Valid @RequestBody PharmacyDto pharmacyDto) {
     }
 
     @DeleteMapping("/{id:[\\d]+}")
-    public ResponseEntity<?> delete(@PathVariable Long id) {
+    public ResponseEntity<Object> delete(@PathVariable Long id) {
```
Comment (Oleksandr-Eleks) [inline api/src/main/java/com/eleks/academy/pharmagator/controllers/PharmacyController.java:51]: I would suggest to use PharmacyDto here

File: api/src/main/java/com/eleks/academy/pharmagator/controllers/PriceController.java [hunk 3]
```diff
@@ -48,7 +48,7 @@
     }
 
     @DeleteMapping("/pharmacyId/{pharmacyId:[\\d]+}/medicineId/{medicineId:[\\d]+}")
-    public ResponseEntity<?> delete(
+    public ResponseEntity<Object> delete(
```
Comment (Oleksandr-Eleks) [inline api/src/main/java/com/eleks/academy/pharmagator/controllers/PriceController.java:51]: I would suggest to use PriceDto here

Comment (zyoneal) [reply api/src/main/java/com/eleks/academy/pharmagator/controllers/PriceController.java]: done

File: api/src/main/java/com/eleks/academy/pharmagator/controllers/PharmacyController.java (hunk 2 above)
Comment (zyoneal) [reply api/src/main/java/com/eleks/academy/pharmagator/controllers/PharmacyController.java]: done

File: api/src/main/java/com/eleks/academy/pharmagator/controllers/MedicineController.java (hunk 1 above)
Comment (zyoneal) [reply api/src/main/java/com/eleks/academy/pharmagator/controllers/MedicineController.java]: done

**Review by kolyaratishin — APPROVED** [review]

**Review by MuchosChuchos — APPROVED** [review]
nice piece of work

**Review by Oleksandr-Eleks — APPROVED** [review]

**Review by sergvasylchak — APPROVED** [review]

**Pull request merged**